### ISSUE TITLE
Wrong results for dense (n>1,m) times sparse (m,1)

### ISSUE THREAD

**Issue by simonwm** [flatironinstitute/sparse_dot#13]
Thanks for sharing this great project! It is 2-4 times faster than anything I could cook up with Numba and usually works like a charm.

Except for one corner case which gives wrong results and sometimes segfaults...

According to my experiments this happens only(?) when a dense matrix of shape (n>1,m) is multiplied with a sparse matrix of size (m,1). The mimimal non-working example
```
import sparse_dot_mkl
import scipy.sparse
import numpy as np

A = np.array([[1.,1.],[1.,1.]])
B = scipy.sparse.csr_matrix([[1.,],[1.,]])

print("sparse_dot_mkl:", sparse_dot_mkl.dot_product_mkl(A, B).flatten())
print("scipy:", (A @ B).flatten())

print('\nversions')
import pkg_resources
import sys
print('sparse-dot-mkl', pkg_resources.get_distribution('sparse-dot-mkl').version)
print('scipy', pkg_resources.get_distribution('scipy').version)
print('numpy', pkg_resources.get_distribution('numpy').version)
print('mkl', sparse_dot_mkl.get_version_string())
print('python', sys.version)
```
prints 
```
sparse_dot_mkl: [2. 0.]
scipy: [2. 2.]

versions
sparse-dot-mkl 0.7.0
scipy 1.5.2
numpy 1.19.5
mkl Intel(R) Math Kernel Library Version 2020.0.4 Product Build 20200917 for Intel(R) 64 architecture applications
python 3.6.10 |Anaconda, Inc.| (default, Mar 25 2020, 23:51:54)
[GCC 7.3.0]
```

Maybe it has something to do with the solution of issue #1 ?

**Comment by simonwm** [comment]
PS: It does also happen for
```
A = scipy.sparse.csr_matrix(np.ones((1,2)))
B = np.ones((2,2)).T
```
but not for
```
A = scipy.sparse.csr_matrix(np.ones((1,2)))
B = np.ones((2,2))
```
or for
```
A = scipy.sparse.csr_matrix(np.ones((1,2)))
B = np.ones((2,2)).T.copy()
```
I assume it is the same bug and the same code is called here for the transposed version due to more efficient use of the memory layout?

**Comment by asistradition** [comment]
Good catch - the numpy memory ordering is ambiguous when one of the dimensions is 1 (as it was in the output array), and it was defaulting to row-major order. It would segfault whenever MKL returned column-major order with the wrong leading dimension.

I fixed it in 7ccb4fa3bf4b4e581a117dbc735c726e0f62da49 and I'll package it for release today.

Thanks for the heads up.

**Comment by asistradition** [comment]
Packaged and released as v0.7.1 - let me know if there are any problems.

**Comment by simonwm** [comment]
Awesome! Thanks a lot!
